### PR TITLE
Modificações

### PR DESCRIPTION
**Pull request by matheus-frota**
# Pull Request
* Implementando Feature Importance
* Renomeando função da matriz de confusão
* Adicionando índice e links para navegação no modelo

### CODE REVIEW

**Comment by ryukinix** [comment]
Os conflitos ocorreram por reautilizar uma branch que havia sido dada merge. Nas futuras modificações como sempre faça o seguinte:

```
git checkout master # entre na master
git pull # atualize a master
git checkout -b new-branch
git add files... # files
git push -u origin new-branch # push to new branch
# if you have hub command do this:
hub pull-request
```

Sobre a ferramenta [hub](https://github.com/github/hub)

**Comment by ryukinix** [comment]
Acredito que ainda há correções necessárias nessa branch como comentado em #10

**Comment by ryukinix** [comment]
O nome desse pull-request não é bom e nem descritivo. Recomendo repensar nas próximas vezes que criar. Exemplo de nome de PR melhor:

+ Adiciona feature importance e outras melhorias

File: asd/metrics.py [hunk 1]
```diff
@@ -50,3 +52,20 @@ def corr_pearson(x, y):
     term1 = np.sum(d1 * d2)
     term2 = np.sqrt(np.sum(d1 ** 2) * np.sum(d2 ** 2))
     return term1 / term2
+
+
+def confusion_performance(y_test, result):
```
Comment (ryukinix) [inline asd/metrics.py]: Acredito que `classifcation_report` seria um nome melhor. Além disso renomeie `result` para `y_pred` que é um nome mais específico.

File: asd/metrics.py [hunk 2]
```diff
@@ -50,3 +52,20 @@ def corr_pearson(x, y):
...
+    """Calcula as métricas de avaliação de uma matriz de confusão"""
+
+    # Extraindo valores da matriz confusão e criando dicionário
+    tn, fp, fn, tp = confusion_matrix(y_test, result).ravel()
```
Comment (ryukinix) [inline asd/metrics.py]: Provavelmente nessa linha há um bug. Como você tem certeza que `fn` e `tp` estão na ordem correta?

File: requirements.txt [hunk 3]
```diff
@@ -2,8 +2,9 @@
 fire>=0.1.3
 
 ## Data analysis
-# pandas
-# numpy
+pandas
+numpy
+scikit-learn
```
Comment (ryukinix) [inline requirements.txt]: Cite ao menos uma versão que está funcionando no seu ambiente, exemplo:

`scikit-learn>=X.Y.Z`. Para verificar a versão atual utilize:

`pip show scikit-learn`

Comment (ryukinix) [reply requirements.txt]: O mesmo conselho para outras bibliotecas.

**Review by ryukinix — CHANGES_REQUESTED** [review]
Algumas sugestões sobre o código além da necessidade de resolver os conflitos no git.

**Comment by ryukinix** [comment]
Fiz algumas correções no código, notebook e também resolvi os conflitos @matheus-frota

**Comment by ryukinix** [comment]
Ponto de atenção sobre o cross_val_predict: 
image


> It is not appropriate to pass these predictions into an evaluation metric. Use cross_validate to measure generalization error.

É necessário como falei anteriormente ter as métricas individuais por modelo. Acredito que você está fazendo uma espécie de Data Leakage aqui passando todo o dataset de treinamento como teste. Como ele irá fazer N estimações ali pelos folds, para alguns modelos a amostra de teste terá sido observada no treinamento.

ref: https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.cross_val_predict.html

**Comment by ryukinix** [comment]
Dado que esse PR está muito grande, criarei uma tarefa a parte discutindo esse problema do cross validation predict + avaliação de métricas.

**Review by ryukinix — APPROVED** [review]
